pkg/pattern/tokenization/replacer.go

```diff
@@ -0,0 +1,859 @@
+		return r.handleSaneTimestamp(hasMinusPrefix, n1, b1)
+
+	// Weird RFC822 dates like "02 Jan 06 15:04 MST"
+	case n1 <= 31 && l1 <= 2 && b1 == ' ':
+		if r.advanceMonthName() && r.advanceChar(' ') && r.advanceYear() && r.advanceChar(' ') && r.advanceTime(true) && r.advanceStringOrNumericTimeZone(false) {
+			r.commit()
+			r.emit(hasMinusPrefix, placeholderTimestamp)
+			return true
+		}
+		// if not, go to default handler after switch statement
+
+	// It could be a UUID that starts with 8 digits
+	case l1 == 8 && b1 == '-':
+		if r.advanceUUIDAfterFirstDash('a', 'f') || (r.resetHeadExpr() && r.advanceChar('-') && r.advanceUUIDAfterFirstDash('A', 'F')) {
+			r.emit(hasMinusPrefix, placeholderUUID)
+			return true
+		}
+		// if not, go to default handler after switch statement
+	}
+
+	// Number with an unknown boundary - emit the number and leave the boundary
+	// for the following passes.
+	r.resetHead()
+	r.emitNumberOrCopyText(hasMinusPrefix)
+	return true
+}
+
+var longDayNames = [...][]byte{
+	[]byte("Sunday"),
+	[]byte("Monday"),
+	[]byte("Tuesday"),
+	[]byte("Wednesday"),
+	[]byte("Thursday"),
+	[]byte("Friday"),
+	[]byte("Saturday"),
+}
+
+func (r *replacer) handleWeirdTimestamp() (endsWithBoundary bool) {
+	r.resetHead()
+
+	if r.advanceDayOfTheWeek() {
+		r.commit() // we will always consume this
+
+		// RFC1123 and RFC1123Z, e.g.:
+		// - "Mon, 02 Jan 2006 15:04:05 MST"
+		// - "Mon, 02 Jan 2006 15:04:05 -0700"
+		if r.advanceChar(',') && r.advanceChar(' ') && r.advanceUintUpTo(31, 2) && r.advanceChar(' ') && r.advanceMonthName() && r.advanceChar(' ') && r.advanceYear() && r.advanceChar(' ') && r.advanceTime(false) && r.advanceStringOrNumericTimeZone(false) {
+			r.emit(false, placeholderTimestamp)
+			return true
+		}
+		r.resetHead()
+
+		// ANSIC, UnixDatem, RubyDate e.g
+		// - "Mon Jan  2 15:04:05 2006"
+		// - "Mon Jan  2 15:04:05 MST 2006"
+		// - "Mon Jan 02 15:04:05 -0700 2006"
+		if r.advanceChar(' ') && r.advanceMonthName() && r.advanceChar(' ') && (r.advanceChar(' ') || true) && r.advanceUintUpTo(31, 2) && r.advanceChar(' ') && r.advanceTime(false) && r.advanceStringOrNumericTimeZone(true) && r.advanceChar(' ') && r.advanceYear() {
+			r.emit(false, placeholderTimestamp)
+			return true
+		}
+		r.resetHead()
+
+		// Linux, e.g.
+		// - "Mon  2 Jan 15:04:05 MST 2006"
+		// - "Tue 23 Jan 15:04:05 -0700 2023"
+		if r.advanceChar(' ') && (r.advanceChar(' ') || true) && r.advanceUintUpTo(31, 2) && r.advanceChar(' ') && r.advanceMonthName() && r.advanceChar(' ') && r.advanceTime(false) && r.advanceStringOrNumericTimeZone(false) && r.advanceChar(' ') && r.advanceYear() {
+			r.emit(false, placeholderTimestamp)
+			return true
+		}
+		r.resetHead()
+
+		// RFC850, e.g.
+		//  - "Monday, 02-Jan-06 15:04:05 MST"
+		backtrackedSlice := r.source[r.head-3:]
+		var matchedDay []byte
+		for _, dw := range longDayNames {
+			if bytes.HasPrefix(backtrackedSlice, dw) {
+				matchedDay = dw
+				break
+			}
```

Comment:
If `longDayNames` was a `map[[3]byte]`, the lookup might be a bit faster, though not sure if that matters a lot for a slice of 7 elements :sweat_smile: Still, while this specific part of the code is probably executed fairly rarely, it couldn't hurt to benchmark :sweat_smile: